### ISSUE TITLE
Setting sheets for web publishing with Google Drive API avoiding CORS issues

### ISSUE THREAD

**Issue by enricofer** [jsoma/tabletop#165]
With reference to https://github.com/jsoma/tabletop/issues/150 and https://github.com/jsoma/tabletop/issues/160:

Following the instruction of issue #150  I can **manually** enable web publishing from google sheets page and bypass cors issues

I found how to get the same result setting permissions from **google drive API** creating a permission with `https://developers.google.com/drive/api/v3/reference/permissions/create` including in request_body the attributes  `"role":"reader", "type":"anyone", "allowFileDiscovery": false` ( getting a `"anyoneWithLink"` id permission)
and setting the head revision (the last) to published with` https://developers.google.com/drive/api/v3/reference/revisions` including in request body the attributes  `"published": true, "publishAuto": true, "publishedOutsideDomain": true,`

![Screenshot 2019-05-13 12 53 49](https://user-images.githubusercontent.com/7544963/57616584-cd2b3200-757e-11e9-9595-98001c587744.png)

**Comment by SarokuTuli** [comment]
Hello,

I tried for the first time Tabletop yesterday and it was working fine. Then I swapped the url of my google spreadsheet and since then I get the infamous '...has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.'

I am not sure what you're doing here and it's not clear to me if it's a "solution", but either Google changed something yesterday afternoon exactly at the moment when I changed my url (improbable) or something else is not working.

I can also see other people have stumble upon this previously, so I am now surprised it actually worked for me in the first place.

Does Tabletop needs to be updated?
